js/src/elements/form-behavior.ts

```diff
@@ -675,10 +683,16 @@ function setBusySubmitButton(event: Event, form: HTMLFormElement, isLoading: boo
 }
 
 export function getLabel(input: HTMLValueElement): string {
-	const label =
-		input.labels && input.labels.length
-			? input.labels[0].textContent
-			: input.getAttribute('aria-label');
+	let label: string | null = null;
+
+	if (input.type === 'radio') {
+		label = input.getAttribute('data-error-label');
```

Comment:
just curious - are we not supporting the retrieval of `aria-label` for radio inputs? Also, is the name `data-error-label` accurately representing the values that can be assigned to it? In your example below, it seems like you are wanting to assign a descriptive name to the input. If that's the case, shouldn't we just call it `data-validation-label` or something?

Reply:
Yeah validation-label sounds like a better name.

As for the aria label, @BenDMyers correct me if Im wrong here, I believe the aria-label is specific to the inputs themselves.  Not the Group.  So with the separate data-attr, we are separating the label to the group rather than the individual input the aria-label would be associated to.

Reply:
Updated label name